src/translators/crate.py

```diff
@@ -313,10 +313,9 @@ def _preprocess_values(self, e, col_names, fiware_servicepath):
                         values.append([float(lon), float(lat)])
                     else:
                         values.append(e[cn]['value'])
-                except KeyError as e:
-                    msg = ("Seems like not all entities of same type came "
-                           "with the same set of attributes. {}").format(e)
-                    raise NotImplementedError(msg)
+                except KeyError:
+                    # this entity update does not have a value for the column so use None which will be inserted as NULL to the db.
+                    values.append( None )
```

Comment:
not sure about the implications of this change? @taliaga @c0c0n3 ?

Reply:
I think this is actually a good change. In fact, you get a `KeyError` when accessing a key that's not in a dictionary. In our case that can happen only if the attribute dictionary has no `value` key, so setting it to `None` which will ultimately result in a DB `null` for the corresponding field is actually consistent with what we've done in PR #122. Also, PR #122 should make this code block unreachable actually, in other words I think the `except` clause is dead code!

Reply:
With my modifications this except clause is not dead code. Its required for example when inserting the new sameTypeEntitiesWithDifferentAttrs test entities I added to src/conftest. There the first entity has attributes temperature and pressure and the second only temperature. This is different from the situation in pull request #122 where the attribute is present but has no value where as here the attribute pressure is completely missing from the entity update. So when _preprocess_values processes the latter entity update the KeyError exception is raised and None is added as the value for pressure. Of course None could be added earlier in src/reporter/reporter.py as its done with missing values but I did it here since it was a simple change.

Reply:
oh right! thanks for clarifying I missed that.